### PR TITLE
[DRL4AMR] Add metadata arrays and new test function

### PR DESCRIPTION
**Pull request by rw-anderson**
This adds a couple new features:
1.  Adds the element id and level number metadata arrays.
2.  Adds a new test oscillatory test function which shows a case where a zz estimator will fail.

### CODE REVIEW

**Comment by jakubcerveny** [comment]
Oh, I realized you started with JS's branch, not mine... mine is much simplified and the ID rasterization can just be added to GetImage().

**Comment by jakubcerveny** [comment]
Or a similar function to GetImage can be added to do ID rasterization.

**Comment by rw-anderson** [comment]
Right, I am reworking this in a branch starting from the point your nice changes were merged in.

**Comment by jakubcerveny** [comment]
Do you think you could merge in the simple version of GetImage? The class `NCM` should really be forgotten...

**Comment by rw-anderson** [comment]
I think it's easier to drop this branch and add my changes starting from the simplified GetImage.  Look for a new branch to replace this one, hopefully today.

**Comment by camierjs** [comment]
I did some simplifications to use Jakub's only way of refining geometries.
We only have one image size, which should be easier for the RL side.
I added a way to use an external seed.

File: miniapps/drl4amr/drl4amr.cpp [hunk 1]
```diff
@@ -12,21 +12,109 @@ using namespace mfem;
...
+      return x0_zzfail(x);
+      break;
+   }
+   printf("unknown function: %d\n",test_func);
+   return 0;
+}
+
+
+// *****************************************************************************
+class NCM: public NCMesh
```
[the remark this answers is not in the document]
Comment (camierjs) [reply miniapps/drl4amr/drl4amr.cpp]: It has been removed, I think it's outdated code you're looking at.

**Pull request merged**